akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -799,7 +874,40 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
 
   def matchingRole(m: Member): Boolean = role.forall(m.hasRole)
 
-  def receive = normalReceive
+  override val supervisorStrategy = {
+    def fromDurableStore: Boolean = sender() == durableStore && sender() != context.system.deadLetters
+    OneForOneStrategy()(
+      ({
+        case e @ (_: DurableStore.LoadFailed | _: ActorInitializationException) if fromDurableStore ⇒
+          log.error(e, "Stopping distributed-data Replicator due to load or startup failure in durable store")
+          context.stop(self)
+          SupervisorStrategy.Stop
+      }: SupervisorStrategy.Decider).orElse(SupervisorStrategy.defaultDecider))
+  }
+
+  def receive =
+    if (hasDurableKeys) load.orElse(normalReceive)
+    else normalReceive
+
+  val load: Receive = {
+    case LoadData(data) ⇒
+      data.foreach {
+        case (key, d) ⇒
+          val envelope = DataEnvelope(d)
+          write(key, envelope) match {
+            case Some(newEnvelope) ⇒
+              if (newEnvelope.data ne envelope.data)
+                durableStore ! Store(key, newEnvelope.data, None)
+            case None ⇒
+          }
+      }
+    case LoadCompleted ⇒
+      context.become(normalReceive)
+      self ! FlushChanges
+
+    case RemovedNodePruningTick | FlushChanges | GossipTick ⇒
+    case _: Read | _: Write | _: Status | _: Gossip         ⇒
```

Comment:
Hm, should you really ignore Write? Isn't the CRDT property a guarantee that writes like this can be handled?